content/articles/django-celery-tasks/index.md

```diff
@@ -0,0 +1,333 @@
+    <h2>Users List</h2>
+
+    <ul>
+        {% for user in object_list %}
+            <li>{{ user.username }} - {{ user.email }} - {{ user.date_joined }}</li>
+        {% empty %}
+            <li>No users. <a href="{% url 'generate' %}">Generate some random users.</a></li>
+        {% endfor %}
+    </ul>
+   {% endblock %}
+   ```
+  
+3. In the `task` directory create a file named `generate_random_user.html` and add the code snippets below. The below code snippet contains an input field where we will specify the number of random users to generate.
+   
+   ```html
+   {% extends 'task/base.html' %}
+
+   {% block content %}
+    <h2>Generate Random Users</h2>
+    <form method="post">
+        {% csrf_token %}
+        <table>
+            {{ form }}
+        </table>
+        <button type="submit">Submit</button>
+    </form>
+   {% endblock %}
+   ```
+
+#### Form
+
+In the `task`, create a python file named `random_user_form.py` and add the code snippet below:
+
+```python
+from django import forms
+from django.core.validators import MinValueValidator, MaxValueValidator
+
+class GenerateRandomUserForm(forms.Form):
+    total = forms.IntegerField(
+        validators=[
+            MinValueValidator(50),
+            MaxValueValidator(500)
+        ]
+    )
+```
+
+### View
+
+Add the code snippet below into the `views.py` file in the `task`:
+
+```python
+from django.shortcuts import redirect
+
+from .form import GenerateRandomUserForm
+from .task import create_random_user_accounts
+
+# returns a list of generated user accounts
+class UsersListView(ListView):
+    template_name = 'task/user_list.html'
+    model = User
+
+# A page with the form where we can input the number of accounts to generate
+class GenerateRandomUserView(FormView):
+    template_name = 'task/generate_random_user.html'
+    form_class = GenerateRandomUserForm
+
+    def form_valid(self, form):
+        total = form.cleaned_data.get('total')
+        create_random_user_accounts.delay(total)
+        messages.success(self.request, 'We are generating your random users! Wait a moment and refresh this page.')
+        return redirect('users_list')
+```
+
+In the above code snippet, notice that we did not call the `create_random_user_accounts` method instead we called `create_random_user_accounts.delay(total)`. This instructs Celery to perform the task in a background process.
+
+#### URLs
+
+Update the `urls.py` with the code snippet below.
+
+```python
+from django.conf.urls import url
+from django.contrib import admin
+from django.urls import path
+
+from task.views import GenerateRandomUserView, UsersListView
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    url('users/', UsersListView.as_view(), name='users_list'),
+    url('generate/', GenerateRandomUserView.as_view(), name='generate')
+
+]
+```
+
+#### Starting the Celery worker process
+
+Execute the command below to start a Celery worker process.
+ 
+```bash
+$ celery -A celerytask worker -l info
+```
```

Comment:
![image](https://user-images.githubusercontent.com/45763204/125572611-bc31caa1-0f09-47b8-8bca-697b53f019d6.png)

After following all the steps until this point, I'm getting this error. Can you please look into this?

Reply:
You are getting the error since you named the project to celery instead of celerytask as I did in the article.

   ```bash
   django-admin startproject celerytask
   ```

Since you used a different name you should be executing the command:

  ```bash
  celery -A celery worker -l info
  ```

Reply:
@njerikaren can you please mention this so that others don't mistake this as I did. `celerytask` seems like a keyword.